internal/ghmcp/server.go

```diff
@@ -44,25 +48,43 @@ type MCPServerConfig struct {
 }
 
 func NewMCPServer(cfg MCPServerConfig) (*server.MCPServer, error) {
-	ghClient := gogithub.NewClient(nil).WithAuthToken(cfg.Token)
-	ghClient.UserAgent = fmt.Sprintf("github-mcp-server/%s", cfg.Version)
-
-	if cfg.Host != "" {
-		var err error
-		ghClient, err = ghClient.WithEnterpriseURLs(cfg.Host, cfg.Host)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create GitHub client with host: %w", err)
-		}
+	apiHost, err := parseAPIHost(cfg.Host)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse API host: %w", err)
 	}
 
+	// Construct our REST client
+	restClient := gogithub.NewClient(nil).WithAuthToken(cfg.Token)
+	restClient.UserAgent = fmt.Sprintf("github-mcp-server/%s", cfg.Version)
+	restClient.BaseURL = apiHost.baseRESTURL
```

Comment:
I chose to go away from `WithEnterpriseURLS` because there are hosts that are not enterprise but require different URLs e.g. review labs or development environments.

Sadly, `githubv4` doesn't offer quite the same options, so I commented below why that is. The alternative would be to construct the `githubv4.Client { Client: graphql.NewClient... }` but I don't think either is particularly problematic.

Reply:
That's a shame! Might need to propose exporting it.